vm/src/memory/paged_memory.rs

```diff
@@ -104,18 +104,18 @@ impl PagedMemory {
     }
 
     pub fn set_words(&mut self, address: u32, values: &[u32]) -> Result<(), MemoryError> {
-        if address
-            .checked_add(values.len() as u32 * WORD_SIZE as u32)
-            .is_none()
-        {
-            return Err(MemoryError::AddressCalculationOverflow);
-        }
-
         if values.is_empty() {
             return Ok(());
         }
 
-        let end_address = address + values.len() as u32 * WORD_SIZE as u32;
+        let words_len_u32 =
```

Comment:
We freely swap between `u32` and `usize` throughout the codebase, due to the fact that it is a RISC-V32 processor this is essentially always safe. So that's much cleaner and I'd recommend just doing the cast here.